cpp/kiss_icp/core/Threshold.cpp

```diff
@@ -25,28 +25,29 @@
 #include <Eigen/Core>
 #include <cmath>
 
-namespace {
-double ComputeModelError(const Sophus::SE3d &model_deviation, double max_range) {
-    const double theta = Eigen::AngleAxisd(model_deviation.rotationMatrix()).angle();
-    const double delta_rot = 2.0 * max_range * std::sin(theta / 2.0);
-    const double delta_trans = model_deviation.translation().norm();
-    return delta_trans + delta_rot;
-}
-}  // namespace
-
 namespace kiss_icp {
+AdaptiveThreshold::AdaptiveThreshold(double initial_threshold,
+                                     double min_motion_threshold,
+                                     double max_range)
+    : min_motion_threshold_(min_motion_threshold),
+      max_range_(max_range),
+      model_sse_(initial_threshold * initial_threshold),
+      model_error_(0.0),
+      num_samples_(1) {}
+
+void AdaptiveThreshold::UpdateModelDeviation(const Sophus::SE3d &current_deviation) {
```

Comment:
I also did some renaming and reshaping in the `AdaptiveThreshold`, which should improve the understanding. Please @nachovizzo @benemer have a look.